src/shared/dns-domain.c

```diff
@@ -959,6 +959,11 @@ bool dns_srv_type_is_valid(const char *name) {
         return c == 2; /* exactly two labels */
 }
 
+bool dnssd_srv_type_is_valid(const char *name) {
+        return dns_srv_type_is_valid(name) &&
+                (dns_name_endswith(name, "_tcp") || dns_name_endswith(name, "_udp")); /* Specific to DNS-SD. RFC 6763, Section 7 */
```

Comment:
dns_name_endswith() can actually return negative, if the name is not valid, this should really be handled I figure, and be returned as "invalid"